src/backend/parser/parse_expr.c

```diff
@@ -941,6 +956,23 @@ rewrite_scope_identity_call(ParseState *pstate, Node **lexpr, Node **rexpr)
 		col_expr = (Var*) *rexpr;
 		func_expr = (FuncExpr*) *lexpr;
 	}
+	else if (IsA(*rexpr, FuncExpr) &&
+            strstr(get_func_name(((FuncExpr*) (*rexpr))->funcid), "like_escape") != NULL)
```

Comment:
Why are we not checking any condition of lexpr here like above case? If this case is sufficient to prove that this the current query has LIKE clause then why do we need the above block at all? We can just check the rexpr and see whether like_escape function is called or not. If so, then we already have sufficient condition to make sure whether the query contains LIKE ESCAPE or not.

Reply:
We dont care about lexpr in here to check for like escape.